repos/system_upgrade/common/actors/checkmountoptions/libraries/checkmountoptions.py

```diff
@@ -0,0 +1,77 @@
+from leapp import reporting
+from leapp.libraries.stdlib import api
+from leapp.models import StorageInfo
+
+
+def inhibit_upgrade_due_var_with_noexec(mountpoint, found_in_fstab=False):
+    summary = (
+        'Leapp detected that the {0} mountpoint is mounted with the "noexec" option, '
+        'which prevents binaries necessary for the upgrade from being executed. '
+        'The upgrade process cannot continue with {0} mounted using the "noexec" option.'
+    )
+
+    if found_in_fstab:
+        hint = (
+            'Temporarily remove the "noexec" option from {0} entry in /etc/fstab until the system is upgraded, '
+            'and remount the partition without the "noexec" option.'
+        )
+        related_resource = [reporting.RelatedResource('file', '/etc/fstab')]
+    else:
+        hint = 'Remount {0} without the noexec option.'
```

Comment:
@MichalHe  thinking about that, in case we do not discover the noexec in /etc/fstab but the mountpoint is noexec, I am curious how this happen. Regarding upgrades, all mountpoins anywhere on the path `/var/lib/leapp` needs to be in fstab when doing the upgrade (otherwise additional issues are hit during the upgrade) and when we find different options than in fstab, it's very suspicious. Probably the msg should be extended e.g. like: `Remount mentioned partitions without the noexec option and ensure this change is persistent during whole in-place upgrade process.`


Btw, if we discover a mountpoint anywhere between `/` and `/var/lib/leapp` and it's not in fstab (we already investigated such a problem regarding /var/lib/leapp mountpoint), we should inhibit the upgrade as well. But I suggest to keep such an extension for a separate PR.